src/plugins/inputs/msg.js

```diff
@@ -1,9 +1,73 @@
 "use strict";
 
-exports.commands = ["msg", "say"];
+const _ = require("lodash");
+const Chan = require("../../models/chan");
+const Msg = require("../../models/msg");
+
+exports.commands = ["query", "msg", "say"];
+
+function parseInput(cmd, args, chan) {
+	switch (cmd) {
+	case "msg":
+	case "query":
+		return args.shift();
+	default:
+		return chan.name;
+	}
+}
 
 exports.input = function(network, chan, cmd, args) {
-	const target = cmd === "msg" ? args.shift() : chan.name;
+	const target = parseInput(cmd, args, chan);
+
+	if (cmd === "query") {
+		if (target.length === 0) {
+			chan.pushMessage(this, new Msg({
+				type: Msg.Type.ERROR,
+				text: "You cannot open a query window without an argument.",
+			}));
+			return;
+		}
+
+		const query = _.find(network.channels, {name: target});
+
+		if (typeof query !== "undefined") {
+			return;
```

Comment:
I left the same comment in previous PR:

> This return means if there's a query open already, the message will not be sent.

Reply:
oh sorry about that I didn't get it. The correct behavior will be to send the message right  ?

Reply:
Yes. It will be very odd otherwise where first /query will open and send a message, but the next one will do nothing.